src/js/pages/export/components/ExportCard.js

```diff
@@ -11,16 +11,14 @@ export class ExportCard extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-
-    if ( nextProps.completedSelected !== this.props.completedSelected) { //check if the state of the button select completed changed
-
-      if (nextProps.completedSelected && nextProps.published) {                // button selected and the chapter is published, mark them as selected
+    if ( nextProps.completedSelected !== this.props.completedSelected || nextProps.allSelected !== this.props.allSelected) { //check if the state of the button select completed changed
+      if ((nextProps.completedSelected && nextProps.published) || nextProps.allSelected) {                // button selected and the chapter is published, mark them as selected
         this.setState({checked: true});
         if (!this.state.checked) {             // to avoid adding them twice when they already selected
           nextProps.selections(nextProps.id, true);         // send id and number to save them into an array, true is used for saved, false for remove
         }
       }
-      if (!nextProps.completedSelected && nextProps.published) {   // once we check the checkbox
+      if ((!nextProps.completedSelected || (nextProps.completedSelected && !nextProps.published)) && !nextProps.allSelected) {   // once we check the checkbox

```

Comment:
this whole componentWillReceiveProps is not very easy to read. maybe we could change the name of the completedSelected prop to something like `checked` for instance. it made more sense for me when i substituted completedSelected for checked, that might have only been for me, but in general i think we can try and make this easier to follow

Reply:
@natibekele and change allSelected to checkedAll?